src/Future.js

```diff
@@ -10,6 +10,16 @@ function jail(handler, f){
   };
 }
 
+function fork(m, reject, resolve) {
```

Comment:
I agree that there is no need for type checking. If we would want to do that then we should do it everywhere in all types.